BPOR/BPOR.Rms/Utilities/ReferenceGenerator.cs

```diff
@@ -0,0 +1,24 @@
+using System.Text;
+using BPOR.Rms.Utilities.Interfaces;
+using LuhnNet;
+
+namespace BPOR.Rms.Utilities;
+
+public class ReferenceGenerator : IReferenceGenerator
+{
+    private static readonly Random _random = new();
+    
+    public string GenerateReference()
+    {
+        var reference = new StringBuilder(16);
+        Span<int> span = stackalloc int[15];
+        for (var i = 0; i < span.Length; i++)
+        {
+            span[i] = _random.Next(0, 10);
+            reference.Append(span[i]);
```

Comment:
The span is good here, but it is not really being used. This could be optimised.

Reply:
yeah, I was trying to take your feedback on creating an array of numbers but then converting to a string but I found myself looping twice which was a waste and the luhn CalculateCheckDigit expects a string array so was all a bit of a mess

Have reverted to string builder only and we can revisit